src/main/java/com/synopsys/integration/detect/configuration/DetectProperties.java

```diff
@@ -479,7 +479,8 @@ private DetectProperties() {
             .setHelp(
                 "Use this value to enable the various snippet scanning modes. For a full explanation, please refer to the 'Running a component scan using the Signature Scanner command line' section in your Black Duck server's online help. Corresponding Signature Scanner CLI Arguments: --snippet-matching, --snippet-matching-only, --full-snippet-scan.")
             .setGroups(DetectGroup.SIGNATURE_SCANNER, DetectGroup.GLOBAL, DetectGroup.SOURCE_SCAN)
-            .build();
+            .build().deprecateValue(SnippetMatching.FULL_SNIPPET_MATCHING,"The option FULL_SNIPPET_MATCHING will be removed for this property as the current implementation uses registration key which will not be granted for customer use after 31st January, 2024.")
+                .deprecateValue(SnippetMatching.FULL_SNIPPET_MATCHING_ONLY,"The option FULL_SNIPPET_MATCHING_ONLY will be removed for this property as the current implementation uses registration key which will not be granted for customer use after 31st January, 2024.");
```

Comment:
Is it not possible to call .setDeprecated say after the .setHelp and before the .build? I think that might get you what you are after without having to come up with a new mechanism? Not sure if you tried that and it didn't work for some reason.

Reply:
Yes, we can call setDeprecated but seeing the previous examples, I felt that it is used whenever we want to deprecate the whole property and not only some particular values for that property. This is such a case as we are continuing the property but only deprecate the full snippet options.

Reply:
yeah you are right, deprecateValue is the way we have done this in the past I see. Looks like only EnumProperty had this. Makes sense will approve :)